### PR TITLE
Add an API to get a nativeImage from a named NSImage

### LINKED ISSUE #9414: Feature request: Add support for "named" NSImage in Touchbar
* Electron version: latest
* Operating system: MacOS

The TouchBar API (and MacOS in general) let's you pass an `imageNamed` param to NSImage to get the default OS image. For example: `[NSImage imageNamed:@"NSTouchBarGoUpTemplate"];` creates an NSImage with the default up chevron icon on MacOS. 

It would be pretty neat-o to allow TouchBar users to do something like this: 

```javascript
new TouchBarButton({
  icon: nativeImage.createFromMacOSNamedImage('NSTouchBarGoUpTemplate')
})
```

...and maybe that would just no-op with `nativeImage.createEmpty()` on non-MacOS platforms. 

I guess this is really bigger than just TouchBar because this API could benefit `Menu` for example as well, or anywhere else where you use `nativeImage`.

### PR DESCRIPTION
**Pull request by MarshallOfSound**
This is useful for fetching macOS specific images for things like the touchbar.

Fixes #9414

### CODE REVIEW

**Comment by felixrieseberg** [comment]
I think @codebytere was _just_ looking for this feature! Solid idea.

**Comment by felixrieseberg** [comment]
I'm not sure where to add this information, but in order to make cross-platform development easier, we could also mention Segoe MDL2 _somewhere_. @MarshallOfSound any ideas? Happy to make the PR.

https://docs.microsoft.com/en-us/windows/uwp/style/segoe-ui-symbol-font

**Comment by gerges** [comment]
This is incredibly useful! Currently doing this via `electron-named-image` but happy to move to core. IIRC the TouchBar images require manual color inversion at the moment. Even if we don't add that ability to `nativeImage` it might be nice to provide a pure JS example in the `touchBar` docs to prevent someone from bringing in a more heavyweight image processing lib (possibly native) for the sake of that simple task.

File: atom/common/api/atom_api_native_image_mac.mm [hunk 1]
```diff
@@ -6,10 +6,29 @@
...
+  @autoreleasepool {
+    NSImage* image = [NSImage imageNamed:base::SysUTF8ToNSString(name)];
+    if (!image.valid) {
+      return CreateEmpty(args->isolate());
+    }
+
+    CGImageRef ref = [image CGImageForProposedRect:nil context:nil hints:nil];
+    NSBitmapImageRep* rep = [[NSBitmapImageRep alloc] initWithCGImage:ref];
+    [rep setSize:[image size]];
+    NSData* pngData = [rep representationUsingType:NSPNGFileType properties:[[NSDictionary alloc] init]];
```
Comment (poiru) [inline atom/common/api/atom_api_native_image_mac.mm]: Nit: `png_data`

**Comment by MarshallOfSound** [comment]
> we could also mention Segoe MDL2 somewhere

Yeah, would be even cooler if someone could figure out how to render an icon from that font into a nativeImage 🤔 

I also just added the ability to HSL shift the named image so that the default black touch bar images can become white easily 👍

**Comment by ccnokes** [comment]
Very cool to see this make it to core. Thanks!

**Pull request merged**